packages/coderskit/src/components/Steps/Steps.tsx

```diff
@@ -0,0 +1,261 @@
+      ? StepContentActive
+      : props.state === 'pending'
+      ? StepContentPending
+      : StepContentBase;
+
+  return (
+    <StepContent {...props} className={classnames(props.className, 'ck-step-content')}>
+      {props.children}
+    </StepContent>
+  );
+};
+
+export const Step = (props: StepProps) => {
+  const StepWrapper = props.labelLayout === 'vertical' ? StepVertical : StepHorizontal;
+
+  return <StepWrapper className={classnames(props.className, 'ck-step')}>{props.children}</StepWrapper>;
+};
+
+// STEP LABEL
+
+export type LabelLayout = 'vertical' | 'horizontal';
+
+export interface StepLabelProps extends LabelHTMLAttributes<HTMLLabelElement> {
+  state?: StepState;
+  children?: ReactNode;
+}
+
+const StepLabelWrapper = styled.label<StepLabelProps>(props => {
+  const { theme, state } = props;
+  const { fontSizes, fontWeights, lineHeights, colors } = theme;
+
+  return {
+    width: 'fit-content',
+    whiteSpace: 'nowrap',
+    fontSize: fontSizes.body2,
+    lineHeight: lineHeights.body2,
+    fontWeight: fontWeights.regular,
+    color: colors[state === 'active' ? 'primary' : 'fontDisabled'],
+  };
+});
+
+export const StepLabel = ({ children, ...props }: StepLabelProps) => {
+  return (
+    <StepLabelWrapper {...props} className={classnames(props.className, 'ck-step-label')}>
+      <span>{children}</span>
+    </StepLabelWrapper>
+  );
+};
+
+// SEPARATOR
+
+export interface SeparatorProps extends StepContentProps {
+  topOffset?: string;
+}
+
+const Separator = styled.div<SeparatorProps>(props => {
+  const { colors } = props.theme;
+
+  return {
+    height: '1px',
+    width: '100%',
+    marginTop: props.topOffset,
+    backgroundColor: colors.fontDisabled,
+  };
+});
+
+// STEP GROUP
+
+export interface StepsProps extends HTMLAttributes<HTMLDivElement> {
+  children?: ReactNode;
+  activeStep?: number;
+  labelLayout?: LabelLayout;
+}
+
+const StepsWrapper = styled.div<StepsProps>(() => {
+  return {
+    display: 'flex',
+    justifyContent: 'space-between',
+  };
+});
+
+const calculateSeparatorTopOffset = (StepWrapper: ReactElement) => {
+  const stepContent = StepWrapper!.props.children[0];
+  return `${stepContent.props.size / 2 + 10}px`;
+};
+
+export const Steps = ({ children, ...props }: StepsProps) => {
+  const className = classnames(props.className, 'ck-steps');
+
+  // Determine child's state and pass number as a child (only if it's active or pending)
+  const childrenWithState = React.Children.map(children as ReactElement[], (child, index) => {
+    const state = index === props.activeStep ? 'active' : index > props.activeStep! ? 'pending' : undefined;
+
+    if (index >= props.activeStep!) {
+      let number = index + 1;
+      let updatedChildren = React.Children.map(child.props.children as ReactElement[], (innerChild, index) => {
+        if (index === 0) {
+          return React.cloneElement(innerChild as ReactElement, { state: state, children: number });
+        } else {
```

Comment:
Skoro w if powyżej znajduje się return, to else nie jest konieczne. Wystarczy, że zostawisz samą jego zawartość.